src/aurora/pefile.cpp

```diff
@@ -48,57 +48,86 @@ const Archive::ResourceList &PEFile::getResources() const {
 }
 
 Common::SeekableReadStream *PEFile::getResource(uint32 index, bool UNUSED(tryNoCopy)) const {
-	// Convert from the PE cursor group/cursor format to the standalone
-	// cursor format.
+	ResourceList::const_iterator iter;
+	for (iter = _resources.begin(); iter != _resources.end(); ++iter) {
+		if (iter->index == index)
+			break;
+	}
```

Comment:
- If index isn't there in _resources, this will leave iter dangling and invoke UB
- Also, index is == the PE resource ID, no? Then you again have the issue that multiple IDs, but different types will clash